src/metabase/lib/query.cljc

```diff
@@ -78,6 +78,31 @@
   [query :- ::lib.schema/query]
   (can-run query))
 
+(defn add-types-to-fields
+  "Add `:base-type` and `:effective-type` to options of fields in `x` using `metadata-provider`. Works on pmbql fields.
+  `:effective-type` is required for coerced fields to pass schema checks."
+  [x metadata-provider]
+  (mbql.u/replace
+   x
+   [:field
+    (options :guard (every-pred map? (complement (every-pred :base-type :effective-type))))
+    (id :guard integer? pos?)]
```

Comment:
That was by accident. Thank you. I'll add fix to the other backport and open new PR targeting master.

Reply:
It should make some test fail, I guess. (Ideally, even kondo should report an error.)

Reply:
First predicate was used, so instead of instead of positive integer it checked for integer only. I thought test against negative integer would be overkill, but I'll try to come up with linter as a follow up.